qiskit/transpiler/synthesis/aqc/cnot_structures.py

```diff
@@ -0,0 +1,325 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+These are the CNOT structure methods: anything that you need for creating CNOT structures.
+"""
+import logging
+
+import numpy as np
+
+# N O T E: we use 1-based indices like in Matlab. To be updated.
+
+_NETWORK_LAYOUTS = {"sequ", "spin", "cart", "cyclic_spin", "cyclic_line"}
+_CONNECTIVITY_TYPES = {"full", "line", "star"}
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_cnots(num_qubits: int, cnots: np.ndarray) -> bool:
+    """
+    Checks validity of CNOT structure.
+    """
+    assert isinstance(num_qubits, (int, np.int64)) and num_qubits >= 1
+    assert isinstance(cnots, np.ndarray)
+    assert cnots.dtype == np.int64 or cnots.dtype == int
+    assert cnots.shape == (2, cnots.size // 2)
+    # pylint: disable=misplaced-comparison-constant
+    assert np.all(np.logical_and(1 <= cnots, cnots <= num_qubits))  # 1-based!!
+    assert np.all(cnots[0, :] != cnots[1, :])
+    return True
```

Comment:
This implementation raises instead of return `False`. That looks unintuitive to me.

Reply:
Well, this was just a check that all parameters are set correctly. If not, an exception is raised. Anyways, replaced with `if` and `raise` statements. The function is removed.